### PR TITLE
laser engraves and rewrites fountain pen flavor  + adds detailed lore

### PR DESCRIPTION
**Pull request by Hopekz**
I've received a lot of positive feedback for this pen so since it is semi-important now I went ahead and took the time to get out some extended lore for this item and re-did the original description flavor text now that I decided it was made by ARMAT.

To-do maybe :
- [ ] Make the pen resistant to acid to comply with the new lore. (not 100% sure yet but would be funny if the only thing left over from a command member dying is their pen)
- [X] This PR or next but I want to add a laser engraving with the owner's name directly on the pen.
- [ ] Probably future PR but now that engravings are here I want to make it so that laser engravings don't happen unless you are command staff

dreamseeker_w51xOzqhf0
dreamseeker_rfZMGoUh14

Update
----
There is now a laser engraving on the owner of the pen. Here is how that looks in-game:

dreamseeker_vUqjoQxujG

:cl: Hopek
add: Changed the description flavortext of the fountain pen to match the new lore. There is now a laser engraving of the owner on the pen itself.
add: Added a detailed lore description for the fountain pen.
/:cl:

### CODE REVIEW

**Comment by Hopekz** [comment]
> You currently have a negative Fix/Feature pull request delta of -21. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

🖕

**Comment by Hopekz** [comment]
> This PR has been inactive for long enough to be automatically marked as stale. This means it is at risk of being auto closed in ~ 7 days, please address any outstanding review items and ensure your PR is finished, if these are all true and you are auto-staled anyway, you need to actively ask maintainers if your PR will be merged. Once you have done any of the previous actions then you should request a maintainer remove the stale label on your PR, to reset the stale timer. If you feel no maintainer will respond in that time, you may wish to close this PR youself, while you seek maintainer comment, as you will then be able to reopen the PR yourself

🖕

File: code/game/objects/items/tools/misc_tools.dm [hunk 1]
```diff
@@ -276,13 +276,29 @@
...
 	icon_state = "fountain_pen"
 	item_state = "fountain_pen"
 	matter = list("metal" = 20, "gold" = 10)
 	var/static/list/colour_list = list("red", "blue", "green", "yellow", "purple", "pink", "brown", "black", "orange") // Can add more colors as required
 	var/current_colour_index = 1
+	var/owner = "hard to read text"
+
+/obj/item/tool/pen/fountain/Initialize(mapload, mob/living/carbon/human/user)
+	. = ..()
+	var/turf/T = get_turf(src)
```
Comment (BeagleGaming1) [inline code/game/objects/items/tools/misc_tools.dm:291]: single letter var

File: code/game/objects/items/tools/misc_tools.dm [hunk 2]
```diff
@@ -276,13 +276,29 @@
...
+	if(locate(/mob/living/carbon/human) in T)
+		var/mob/new_owner = (locate(/mob/living/carbon/human) in T)
+		owner = new_owner.real_name
+	if(locate(/obj/structure/machinery/cryopod) in T)
+		var/obj/structure/machinery/cryopod/new_owners_pod = (locate(/obj/structure/machinery/cryopod) in T)
+		owner = new_owners_pod.occupant.real_name
+
+/obj/item/tool/pen/fountain/get_examine_text(mob/user)
+	. = ..()
+	. += "There's a laser engraving of [owner] on it."
```
Comment (BeagleGaming1) [inline code/game/objects/items/tools/misc_tools.dm:301]: what if there is no owner to it?

Comment (Hopekz) [reply code/game/objects/items/tools/misc_tools.dm:301]: That's handled on line 287

So it becomes "There's a laser engraving of hard to read text on it."

File: code/game/objects/items/tools/misc_tools.dm (hunk 1 above)
Comment (Hopekz) [reply code/game/objects/items/tools/misc_tools.dm]: fixed

File: code/game/objects/items/tools/misc_tools.dm [hunk 3]
```diff
@@ -276,13 +276,29 @@
...
+
+/obj/item/tool/pen/fountain/Initialize(mapload, mob/living/carbon/human/user)
+	. = ..()
+	var/turf/current_turf = get_turf(src)
+	if(locate(/mob/living/carbon/human) in current_turf)
+		var/mob/new_owner = (locate(/mob/living/carbon/human) in current_turf)
+		owner = new_owner.real_name
+	if(locate(/obj/structure/machinery/cryopod) in current_turf)
+		var/obj/structure/machinery/cryopod/new_owners_pod = (locate(/obj/structure/machinery/cryopod) in current_turf)
+		owner = new_owners_pod.occupant.real_name
```
Comment (harryob) [inline code/game/objects/items/tools/misc_tools.dm:297]:
```suggestion
	var/mob/living/carbon/human/new_owner = locate() in current_turf
	if(new_owner)
		owner = new_owner.real_name
	var/obj/structure/machinery/cryopod/new_owners_pod = locate() in current_turf
	if(new_owners_pod)
		owner = new_owners_pod.occupant.real_name
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Hopekz** [comment]
Might reopen this in the future when CM dev picks back up closer to how it was when I started. (turnaround back to 2-4 days)

Personally for me it is a bit mentally exhausting at the moment to be open to modify something for a month so assuming I haven't moved on to another project I'll reopen this when I come back to CM dev.

**Comment by Hopekz** [comment]
On second review this one is probably fine too

File: code/game/objects/items/tools/misc_tools.dm [hunk 4]
```diff
@@ -276,13 +276,29 @@
...
+
+/obj/item/tool/pen/fountain/Initialize(mapload, mob/living/carbon/human/user)
+	. = ..()
+	var/turf/current_turf = get_turf(src)
+	var/mob/living/carbon/human/new_owner = locate() in current_turf
+	if(new_owner)
+		owner = new_owner.real_name
+	var/obj/structure/machinery/cryopod/new_owners_pod = locate() in current_turf
+	if(new_owners_pod)
+		owner = new_owners_pod.occupant.real_name
```
Comment (harryob) [inline code/game/objects/items/tools/misc_tools.dm:297]:
```suggestion
		owner = new_owners_pod.occupant?.real_name
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**